content/cli/workspaces.md

```diff
@@ -0,0 +1,206 @@
+
+All workspaces have a **default application**.  In the `nest-cli.json` file above, the default project is `my-project`.  This is defined by the top-level `"root"` property, which points at the root of the default project.  Usually, this is the *single project* mode application you started with, and later added applications with `nest generate app`.  Default projects are used by `nest` commands like `nest build` and `nest start` when a project name is not supplied.
+
+For example, in the above structure, running
+```bash
+$ nest start
+```
+
+Will start up the `my-project` app.  To start `my-app`, we'd do:
+
+```bash
+$ nest start my-app
+```
+
+#### Workspace global properties
+
+Workspaces have additional workspace-wide properties that help hide the complexity of working in a multi-project environment.  These are reflected in the following top-level properties of the `nest-cli.json` file:
+
+- collection: points at the collection of schematics used to generate components; you generally should not change this value
+- sourceRoot: points at the root of the source code for the *default project*
+- monorepo: for a workspace mode structure, this value is always `true`
+- root: points at the project root of the *default project*
+- compilerOptions: a map with keys specifying compiler options and values specifying the option setting
+
+#### Compiler options
+
+Notice that by default, `compilerOptions` is set as follows:
+```json
+  "compilerOptions": {
+    "webpack": true,
+    "tsConfigPath": "apps/my-project/tsconfig.app.json"
+  }
+```
+
+The `webpack` property, when true (the default), uses [webpack]() to compile and bundle the code.  This is one key difference between *workspace mode* and *single project mode*.  *Single project mode* uses `tsc` for compilation by default.  The reason for this difference is that webpack can have significant advantages in build times and in producing a single file bundling all project components together.  If you wish to generate individual files, set `webpack` to `false`, which will cause the build process to use `tsc`.
+
+**Note to Kamil**: can you please explain the significance of why `compilerOptions.tsConfigPath` is set to the default project?
+
+### Projects
+
+As mentioned, workspaces are composed of projects, which may be of the **application** or **library** type.  These are discussed below.
+
+#### Applications
+
+As mentioned, application-type projects, or what we might informally refer to as just "applications", are complete Nest applications that you can run and deploy.  You generate an application-type project with `nest generate app`.
+
+> info **Notice** technically, these application-type projects are called `sub-apps`.  `sub-app` is the name of the schematic you run.  The `sub-app` schematic has an alias of `app`, so you can run either `nest generate app` or `nest generate sub-app`.
```

Comment:
`sub-app` is a private name that can change eventually (to just `g application`) so we should stick to `nest g app` in the docs